### PR TITLE
Rework Airflow images

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

This brings the Airflow Dockerfile in line with the latest changes for Hadoop and applies some lessons we learned and adds some new comments.

In particular:
- Moves git_sync, tini and statsd_exporter versions to variables
- Gets everything from our own repository
- Syncs entrypoint.sh from airflow upstream

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
```

### CODE REVIEW

File: airflow/stackable/utils/entrypoint.sh [hunk 1]
```diff
@@ -290,7 +311,7 @@
     >&2 echo "         the container starts, so it is onlny useful for testing and trying out"
     >&2 echo "         of adding dependencies."
     >&2 echo
-    pip install --no-cache-dir ${_PIP_ADDITIONAL_REQUIREMENTS}
+    pip install --root-user-action ignore --no-cache-dir ${_PIP_ADDITIONAL_REQUIREMENTS}
```
[the remark this answers is not in the document]
Comment (lfrancke) [reply airflow/stackable/utils/entrypoint.sh:314]: Ignoring this as it comes from upstream

File: airflow/Dockerfile [hunk 2]
```diff
@@ -10,33 +20,33 @@ COPY airflow/constraints-${PRODUCT}-python${PYTHON}.txt /tmp/constraints.txt
...
+        python${PYTHON}-pip \
+        python${PYTHON}-wheel \
+        unixODBC-devel && \
+    microdnf clean all && \
+    python3 -m venv --system-site-packages /stackable/app && \
+    source /stackable/app/bin/activate && \
+    pip install --no-cache-dir --upgrade pip && \
+    pip install --no-cache-dir apache-airflow[${AIRFLOW_EXTRAS}]==${PRODUCT} --constraint /tmp/constraints.txt && \
+    # Needed for pandas S3 integration to e.g. write and read csv and parquet files to/from S3
+    pip install --no-cache-dir s3fs
```
Comment (sbernauer) [inline airflow/Dockerfile:41]: I would say we need s3fs in the final image, not the builder. Hupsi :innocent:

File: hadoop/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,12 @@
 [#347]: https://github.com/stackabletech/docker-images/pull/347
 [#400]: https://github.com/stackabletech/docker-images/pull/400
 
+### Changed
+
+- Upgraded to JMX Exporter 0.19 ([#400])
```
Comment (sbernauer) [inline hadoop/CHANGELOG.md:19]: Please move it into existing `Changed` section and change number to 431

**Review by sbernauer — CHANGES_REQUESTED** [review]
LGTM otherwise

File: airflow/Dockerfile (hunk 2 above)
Comment (lfrancke) [reply airflow/Dockerfile:41]: Oh, sorry! I must have moved this accidentally. I'll move it back.

File: hadoop/CHANGELOG.md (hunk 3 above)
Comment (lfrancke) [reply hadoop/CHANGELOG.md:19]: Uhm...this whole thing was accidental. I'll check why this is in here.

File: airflow/Dockerfile (hunk 2 above)
Comment (lfrancke) [reply airflow/Dockerfile:41]: I've looked. I think this is actually correct.
This is built/installed in a Virtualenv and we move the whole directory over to the final image later.

File: hadoop/CHANGELOG.md (hunk 3 above)
Comment (lfrancke) [reply hadoop/CHANGELOG.md:19]: This is actually also correct. This is the Hadoop changelog...but you're correct in that the Airflow changelog hasn't been updated.

File: airflow/CHANGELOG.md [hunk 4]
```diff
@@ -2,21 +2,39 @@
 
 ## [Unreleased]
 
+### Changed
+
+- Updated Vector to 0.31 ([#429]).
+- Updated git-sync to 3.6.8 ([#431]).
+- Updated statsd-exporter to 0.24, this was accidentally moved to a very old version previously (0.3.0) ([#431])
```
Comment (sbernauer) [inline airflow/CHANGELOG.md:9]:
```suggestion
- Updated statsd-exporter to 0.24, this was accidentally moved to a very old version previously (0.3.0) ([#431]).
```

File: airflow/CHANGELOG.md [hunk 5]
```diff
@@ -2,21 +2,39 @@
...
+
+[#429]: https://github.com/stackabletech/docker-images/pull/429
+[#431]: https://github.com/stackabletech/docker-images/pull/431
+
+## [23.7.0]
+
+NOTE: Some of these changes might have been released in earlier versions already, we just forgot to update the changelog
+
```
Comment (sbernauer) [inline airflow/CHANGELOG.md:17]:
```suggestion
```

**Review by sbernauer — DISMISSED** [review]

**Review by sbernauer — APPROVED** [review]

**Pull request merged**